### PR TITLE
feat(renderer): added dynamic font loading based on lang requirements

### LINKED ISSUE #2723: [Improvement]: The built-in font causes some Chinese to be displayed as ? logo, is it possible to fix or disable the font that comes with it?
### Which feature

The built-in font causes some Chinese to be displayed as ? logo, is it possible to fix or disable the font that comes with it?

### Reason

The built-in font causes some Chinese to be displayed as ? logo, is it possible to fix or disable the font that comes with it?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Yimura**
Related YimMenu/Translations#117
Closes #2723

Changes made:
 - Renderer class was changed to an inline instance (instead of inline ptr)
 - Some throw code was removed because of change of instantiation of the renderer
 - Fonts are dynamically loaded based on the language requirements
 - renderer files have been moved to make the root directory less messy
 - Added a font mgr to improve the ease of adding more language support

### CODE REVIEW

File: src/renderer/renderer.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		m_wndproc_callbacks.emplace_back(callback);
+	}
+
+	void renderer::on_present()
+	{
+		if (m_font_mgr.rebuild_required())
+		{
+			m_font_mgr.rebuild_now();
+			pre_reset();
+			post_reset();
```
Comment (Yimura) [inline src/renderer/renderer.cpp]: These are pretty heavy, @xiaoxiao921 any ideas on doing it with more lightweight calls?

https://github.com/YimMenu/YimMenu/blob/16600cdd7e128cb27684d7bb4f26ff44fbcb0994/src/renderer/renderer.cpp#L106-L114

**Comment by yousef157** [comment]
off-topic but does the menu render arabic text correctly? might make a translation if i'm not bothered

**Comment by gir489returns** [comment]
> off-topic but does the menu render arabic text correctly? might make a translation if i'm not bothered

I was also thinking of adding a Hebrew translation, but the only people I know who speak Hebrew and don't speak English are fucking boomers, so I CBF'd, all my Isreali friends speak English. However, if you'd like to add an Arabic translation, we can add support for it with this new addition of modular fonts. All you need to do when you submit your translation is tell us a font that comes with Windows that supports your language's typeface.

**Comment by yousef157** [comment]
> > off-topic but does the menu render arabic text correctly? might make a translation if i'm not bothered
> 
> I was also thinking of adding a Hebrew translation, but the only people I know who speak Hebrew and don't speak English are fucking boomers, so I CBF'd, all my Isreali friends speak English. However, if you'd like to add an Arabic translation, we can add support for it with this new addition of modular fonts. All you need to do when you submit your translation is tell us a font that comes with Windows that supports your language's typeface.

I'm not sure how I would translate certain strings anyway, like for example "God mode", whether I should translate it to invincibility in Arabic, or literally just write Allah mode (in arabic), or perhaps God mode in transliteration.... I mean, like, there's a shitton of other strings too, but, I'll think about it

also I'm not sure what you meant by the last sentence

Sorry for not asking on the Matrix btw, I just wasn't able to set it up correctly

**Comment by gir489returns** [comment]
> I'm not sure how I would translate certain strings anyway, like for example "God mode", whether I should translate it to invincibility in Arabic, or literally just write Allah mode (in arabic), or perhaps God mode in transliteration.... I mean, like, there's a shitton of other strings too, but, I'll think about it
> 
> also I'm not sure what you meant by the last sentence
> 
> Sorry for not asking on the Matrix btw, I just wasn't able to set it up correctly

Yeah, we've had this discussion multiple times about words that don't translate well into other languages.  We ultimately leave those choices up the translators, as we hope that them being a 1st hand knowledge base on this would solve the debate. There's a lot of phrases that comes from GTA V, the cheating scene, this project, or a combination of all 3. If you peruse the closed tickets for the Translation repo, you can see the havoc it wreaks. Hebrew has the same problem, there is no word for God, there is only Elohim, which is basically the same as Allah. In fact, it is a sin in Judaism to even speak the lord's name, if you wish to write it out, a lot of Jews type it out as G-d. 

You need to submit a font that allows you to type in Arabic. If you can't find one, then let us know, and we'll do our best to find one.

**Comment by Yimura** [comment]
Haven't been able to crash myself so I consider it ready to merge.

**Review by gir489returns — APPROVED** [review]
I haven’t tested the functionality but the code looks good.

**Pull request merged**